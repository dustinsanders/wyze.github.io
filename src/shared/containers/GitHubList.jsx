// @flow

import type { GitHubResult, GitHubResults, Style, VNode } from '../../types'
import { css } from 'glamor'
import Box from '../components/Box'
import GitHubItem from '../components/GitHubItem'
import preact from 'preact'

type Props = {
  content: GitHubResults,
  title: string,
}

const childStyle: Style = css({
  justifyContent: 'flex-start',
})

const GitHubList = ({ content, title }: Props): VNode => (
  <Box wrap childStyle={childStyle} title={title}>
    {content.map(( item: GitHubResult ): VNode => (
      <GitHubItem key={item.name} {...item} />
    ))}
  </Box>
)

export default GitHubList
