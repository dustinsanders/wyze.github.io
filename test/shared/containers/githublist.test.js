// @flow

import GitHubList from '../../../src/shared/containers/GitHubList'
import snapshot from '../../helpers/snapshot'

const ghItemWithoutDescription = {
  name: 'wyze.github.io',
  stars: 1,
  url: 'https://github.com/wyze/wyze.github.io',
}

const ghItem = {
  ...ghItemWithoutDescription,
  description: 'My personal website.',
}

const content = [
  ghItemWithoutDescription,
  ghItem,
]

const title = 'Open Source Projects'

describe('<GitHubList />', () => {
  it('renders', snapshot.bind(null, GitHubList, { content }))
  it('renders title', snapshot.bind(null, GitHubList, { content, title }))
})
