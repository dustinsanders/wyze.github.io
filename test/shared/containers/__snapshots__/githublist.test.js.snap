// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GitHubList /> renders 1`] = `
"<Box wrap childStyle=\\"css-1bslz89\\">
  <GitHubItem name=\\"wyze.github.io\\" stars=\\"1\\" url=\\"https://github.com/wyze/wyze.github.io\\"></GitHubItem>
  <GitHubItem name=\\"wyze.github.io\\" stars=\\"1\\" url=\\"https://github.com/wyze/wyze.github.io\\" description=\\"My personal website.\\"></GitHubItem>
</Box>"
`;

exports[`<GitHubList /> renders title 1`] = `
"<Box wrap childStyle=\\"css-1bslz89\\" title=\\"Open Source Projects\\">
  <GitHubItem name=\\"wyze.github.io\\" stars=\\"1\\" url=\\"https://github.com/wyze/wyze.github.io\\"></GitHubItem>
  <GitHubItem name=\\"wyze.github.io\\" stars=\\"1\\" url=\\"https://github.com/wyze/wyze.github.io\\" description=\\"My personal website.\\"></GitHubItem>
</Box>"
`;
